src/sonic-config-engine/sonic_device_util.py

```diff
@@ -44,6 +44,32 @@ def get_npu_id_from_name(npu_name):
     else:
         return None
 
+def get_npu_pci_device_id(npu_id):
+    platform = get_platform_info(get_machine_info())
+    if not platform:
+        return None
+
+    asic_conf_file_path = os.path.join(SONIC_DEVICE_PATH, platform, ASIC_CONF_FILENAME)
+    if not os.path.isfile(asic_conf_file_path):
+        return None
+
+    # In a multi-npu device we need to have the file "asic.conf" updated with the asic instance
+    # and the corresponding pci id. Below is an eg: for a 2 ASIC platform/sku.
+    # PCI_ID_ASIC_0=03:00.0
+    # PCI_ID_ASIC_1=04:00.0
```

Comment:
do we want to add check pci id format need to be x:y.z ?

Reply:
I was preferring to keep the logic to just fetch what the user has added as PCI device id's in the asic.conf file and use it directly to update the "asic_id" field. We might have to validate a bit more than just the format to check if that is a valid PCI device id on this platfrom. So it is to the platform owner to populate the correct PCI device id ?

Reply:
@judyjoseph as discussed i think we should API name to be generic as instance id can be pci_id or something else in some other platforms.

Reply:
I have updated the code to use generic device_id naming convention as discussed. So here in the example I have shown the platform where the device_id is pci_id.